### PR TITLE
MAINT: cleanup setup.py and remove redundant requirements.txt.

### PR DESCRIPTION
**Pull request by cournape**
We don't need a separate `requirements.txt` anymore. I've also cleaned up our `dev_requirements.txt` and `setup.py`.

### CODE REVIEW

**Comment by cournape** [comment]
@johntyree that should clear things up

File: setup.py [hunk 1]
```diff
@@ -1,6 +1,18 @@
...
 
 
+INSTALL_REQUIRES = [
+    "attrs >= 15.2.0",
+    "okonomiyaki == 0.14.0",
+    "six >= 1.9.0"
+]
+
+EXTRAS_REQUIRE = {
+    ':python_version=="2.7"': [
```
Comment (johntyree) [inline setup.py]: This doesn't seem to work. Should I have to do anything other than `pip install -e .` ?

Comment (johntyree) [inline setup.py]: My fault. :hankey:

**Comment by johntyree** [comment]
LGTM!

**Pull request merged**